src/components/connect.js

```diff
@@ -307,15 +307,16 @@ function connect(mapPropsToRequestsToProps, defaults, options) {
         return (meta) => {
           return (reason) => {
             if (mapping.catch) {
-              this.refetchDatum(prop, coerceMapping(null, mapping.catch(reason, meta), mapping))
-              return
+              return this.refetchDatum(prop, coerceMapping(null, mapping.catch(reason, meta), mapping))
             }
 
             this.setAtomicState(prop, startedAt, mapping, PromiseState.reject(reason, meta), null, () => {
               if (mapping.andCatch) {
                 this.refetchDataFromMappings(mapping.andCatch(reason, meta))
               }
             })
+
+            throw reason
```

Comment:
This line I'm not sure about, because up until now the promise created in `createPromise` was handled (ie. became a fulfilled promise) if it rejected. With this change, re-throwing the exception, we're making it an un-handled promise, which will emit the `unhandledRejection` event that browsers are starting to implement. I'm not sure it makes sense for react-refetch, as a library, to leave the promise unhandled like that, especially since we do handle it (set the correct state in the component etc.)